examples/OpenGL45/matrix_multiplication/matrix_multiplication_version2.shadertrap

```diff
@@ -0,0 +1,290 @@
+BIND_SHADER_STORAGE_BUFFER BUFFER array1_mat_buf BINDING 1
+
+RUN_COMPUTE
+    PROGRAM compute_prog_1
+    NUM_GROUPS 16 16 1
+
+ASSERT_EQUAL BUFFERS array1_mat_buf correct_array1_mat_buf
+
+DECLARE_SHADER shader_2 KIND COMPUTE
+#version 450
+#define KERNEL_SIZE 4
+#define MAT_SIZE  16
+
+
+layout(local_size_x=1, local_size_y=1, local_size_z=1) in;
+
+layout(std430, binding = 0) buffer array2 {
+  float[MAT_SIZE][MAT_SIZE] data2;
+};
+layout(std430, binding=1) buffer array2_mat_buf{
+  mat4x4 array2_mat[4][4];
+};
+void main() {
+    uint row = uint(gl_WorkGroupID[0]), col = uint(gl_WorkGroupID[1]);
+    array2_mat[row/KERNEL_SIZE][col/KERNEL_SIZE][col%KERNEL_SIZE][row%KERNEL_SIZE] = data2[row][col];
+}
+END
+
+COMPILE_SHADER shader_compiled_2 SHADER shader_2
+CREATE_PROGRAM compute_prog_2 SHADERS shader_compiled_2
+
+BIND_SHADER_STORAGE_BUFFER BUFFER array2 BINDING 0
+BIND_SHADER_STORAGE_BUFFER BUFFER array2_mat_buf BINDING 1
+
+RUN_COMPUTE
+    PROGRAM compute_prog_2
+    NUM_GROUPS 16 16 1
+
+ASSERT_EQUAL BUFFERS array2_mat_buf correct_array2_mat_buf
+
+
+DECLARE_SHADER shader_3 KIND COMPUTE
+#version 450
+#define KERNEL_SIZE 4
+#define MAT_SIZE  16
+
+layout(local_size_x=1, local_size_y=1, local_size_z=1) in;
+
+layout(std430, binding = 0) buffer array1 {
+  float[MAT_SIZE][MAT_SIZE] data1;
+};
+layout(std430, binding=1) buffer array2 {
+  float[MAT_SIZE][MAT_SIZE] data2;
+};
+layout(std430, binding=2) buffer array1_mat_buf{
+  mat4x4 array1_mat[4][4];
+};
+layout(std430, binding=3) buffer array2_mat_buf{
+  mat4x4 array2_mat[4][4];
+};
+layout(std430, binding=4) buffer result{
+  float[MAT_SIZE][MAT_SIZE] res;
+};
+
+void main() {
+
+    mat4x4 C;
+    uint row = uint(gl_WorkGroupID[0]), col = uint(gl_WorkGroupID[1]);
+    for(int i = 0;i < MAT_SIZE/KERNEL_SIZE; i++){
+        for(int j = 0; j < MAT_SIZE/KERNEL_SIZE; j++){
+            C = mat4x4(0);
+            for(int k = 0; k < MAT_SIZE/KERNEL_SIZE; k++)
+                 // The submatrix at position (i,j) is equal to the multiplication
+                // of all the submatrices (i,k) and (k,j) where k = 0....(MAT_SIZE/KERNEL_SIZE).
+                C += array1_mat[i][k] * array2_mat[k][j];
+            
+            for(int p = 0; p < KERNEL_SIZE; p++){
+                for(int q = 0; q < KERNEL_SIZE; q++){
+                    // Converts the 4x4 array of 4x4 matrices to a 16x16 array
+                    res[i * KERNEL_SIZE + p][j * KERNEL_SIZE + q] = C[q][p];
+                }
+            }
+        }
+    }
+
+}
+END
+
+COMPILE_SHADER shader_compiled_3 SHADER shader_3
+CREATE_PROGRAM compute_prog_3 SHADERS shader_compiled_3
+
+BIND_SHADER_STORAGE_BUFFER BUFFER array1 BINDING 0
+BIND_SHADER_STORAGE_BUFFER BUFFER array2 BINDING 1
+BIND_SHADER_STORAGE_BUFFER BUFFER array1_mat_buf BINDING 2
+BIND_SHADER_STORAGE_BUFFER BUFFER array2_mat_buf BINDING 3
+BIND_SHADER_STORAGE_BUFFER BUFFER result BINDING 4
+
+RUN_COMPUTE
+    PROGRAM compute_prog_3
+    NUM_GROUPS 16 16 1
+
```

Comment:
Once you have changed the compute shader so that it produces a result matrix in blocked form, please do an assertion to check that this blocked form is expected. Then run a final compute shader to unblock the data (and check that it has the final expected result).